site/docs/v1/tech/admin-guide/upgrade.adoc

```diff
@@ -4,44 +4,300 @@ title: Upgrade FusionAuth
+- Start the first node and add it back to the load balancer.
+- Stop the second and third nodes, remove them from the load balancer, and upgrade them. 
+- Start the second and third nodes and add them back to the load balancer.
+
+There is still some downtime with this procedure, but it is limited to the time it takes to run the database migration on the upgraded first node. There will be limited capacity during the upgrade, especially while the second and third nodes are upgrading, but the system will be available for use for most of the upgrade process.
+
+++++
+{% plantuml source: _diagrams/docs/admin-guide/rolling_upgrade.plantuml, alt: "Rolling upgrade." %}
+++++
+
+=== Blue-Green Deployment
+
+A blue-green deployment is a technique that reduces downtime and risk by running two identical environments in parallel. A load balancer is used to direct traffic to one of the environments. Using a blue-green deployment instead of performing a rolling upgrade isolates the upgrade process from the production environment, reducing the impact and risk to the current production environment.
+
+To perform a blue-green deployment with a three-node cluster, first create three new nodes with the new version of FusionAuth on each of them. Start up one of the new nodes. When the new node starts and connects to the database, it will lock the database and own it while it completes the database upgrade. The original nodes might give errors as they will not have access to the database. These original nodes can be removed. Now add the new nodes to the cluster. The system will be down for the duration of the database upgrade but once the database upgrade is complete, the system will be back online with the upgraded version of FusionAuth on the new nodes. 
+
+++++
+{% plantuml source: _diagrams/docs/admin-guide/blue_green_upgrade.plantuml, alt: "Blue green deployment." %}
+++++
+ 
+=== Out-of-Band Database Upgrades
+
+FusionAuth can handle database migrations automatically using link:/docs/v1/tech/guides/silent-mode[Silent Mode]. FusionAuth can also handle updates interactively on development environments using link:/docs/v1/tech/installation-guide/fusionauth-app#maintenance-mode[Maintenance Mode].
+
+
+[WARNING]
+====
+Note that Maintenance Mode should never be used in production.
+
+You should always back up your database before upgrading regardless of the upgrade method you choose.
 ====
 
 
-== ZIP Packages
+In production, you may want to perform the database upgrade manually, also known as an out-of-band upgrade. This is useful if you want to perform the database upgrade in a way that minimizes downtime or as part of a larger automated or coordinated deployment. Upgrading the database manually involves running SQL scripts directly on your database to update the database schema. 
```

Comment:
```suggestion
In production, you may want to perform the database upgrade rather than allow FusionAuth to do so. This is also known as an out-of-band upgrade. This is useful if you want to perform the database upgrade in a way that minimizes downtime or as part of a larger automated or coordinated deployment. Upgrading the database involves running SQL scripts to update the database schema. 
```